src/current/_includes/v23.1/start-in-docker/mac-linux-steps-single-node.md

```diff
@@ -105,21 +104,25 @@ The `cockroach` process listens on `127.0.0.1:26257` and `localhost:26257`, and
 
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    docker log roach-single --follow
+    docker logs --follow roach-single
     ~~~
 
-    To stop monitoring the logs, press Ctrl+C to exit the `docker log` command.
+    To stop monitoring the logs, press Ctrl+C to exit the `docker logs` command.
 
-1. To connect to the cluster interactively using the `cockroach sql` command-line interface, set `--url` cluster's SQL connection string, which is printed next to `sql:` in the cluster's startup details. You can replace the IP address with the hostname of the Docker container so that the script continues to work if the IP address changes. To run `cockroach sql` on the `roach-single` cluster:
+1. To connect to the cluster interactively using the `cockroach sql` command-line interface, set `--url` cluster's SQL connection string, which is printed next to `sql:` in the cluster's startup details. Connect to the `roach-single` cluster:
 
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    docker exec -it roach-single ./cockroach sql --url="postgresql://root@roach-single:26257/defaultdb?sslcert=certs%2Fclient.root.crt&sslkey=certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=certs%2Fca.crt"
+    docker exec -it roach-single ./cockroach sql --url="postgresql://root@127.0.0.1:26257/defaultdb?sslcert=certs%2Fclient.root.crt&sslkey=certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=certs%2Fca.crt"
```

Comment:
Can you revert this change after recent discussion?

Reply:
This change needs to remain. Otherwise users will receive an x509 error when trying to connect to the database:

```sh
$ docker exec -it roach-single ./cockroach sql --url="postgresql://root@roach-single:26257/defaultdb?sslcert=certs%2Fclient.root.crt&sslkey=certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=certs%2Fca.crt"
#
# Welcome to the CockroachDB SQL shell.
# All statements must be terminated by a semicolon.
# To exit, type: \q.
#
ERROR: failed to connect to `host=roach-single user=root database=defaultdb`: failed to write startup message (x509: certificate is not valid for any names, but wanted to match roach-single)
Failed running "sql"
```

Reply:
Oh man, I just noticed something about the `start-single-node` procedure. It's not using `--insecure`. I guess maybe that is disallowed for a single-node cluster. We can put that to the side for now. It would be better to _eventually_ (outside of this PR) update the earlier procedure to not use `--insecure` I think, though it introduces an extra complication.

I further think that the problem here is that the certificate inside the container is generated only for `localhost` and not for the container's name (which is also its DNS hostname), so you're seeing the x509 error. I think this points to the potential for a feature request to create the certificate with all of the in-container IPs it is binding to, and their resolved DNS hostnames. WDYT? cc/ @ZhouXing19

Reply:
> Oh man, I just noticed something about the `start-single-node` procedure. It's not using `--insecure`. I guess maybe that is disallowed for a single-node cluster. We can put that to the side for now. It would be better to _eventually_ (outside of this PR) update the earlier procedure to not use `--insecure` I think, though it introduces an extra complication.

It's not disallowed. I actually had my changes using `--insecure` originally until I saw that the page mentioned "By default, a `certs` directory is created and CockroachDB starts in secure mode." right after the `docker run` command. I figured that it was intentionally starting the cluster in secure mode even though the multi-node cluster instructions above it used `--insecure`. Happy to introduce `--insecure` if it's desired here.

> I further think that the problem here is that the node certificate inside the container is generated only for `localhost` and not for the container's name (which is also its DNS hostname), so you're seeing the x509 error. I think this points to the potential for a feature request to create the certificate with all of the in-container IPs it is binding to, and their resolved DNS hostnames. WDYT? cc/ @ZhouXing19

The TLS certificate that's generated uses `node` as the Common Name (CN) and `127.0.0.1` as IP Subject Alternative Names (SANs). Trying to connect using the `roach-single` name will give x509 errors.

```sh
$ docker cp roach-single:/cockroach/certs/node.crt ./node.pem && openssl x509 -in ./node.pem -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            5d:65:09:26:a9:92:8a:b8:cc:73:3a:7d:70:af:50:07
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: O = Cockroach, CN = Cockroach CA
        Validity
            Not Before: Nov 14 22:19:55 2023 GMT
            Not After : Nov 18 22:19:55 2028 GMT
        Subject: O = Cockroach, CN = node
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    REDACTED
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage:
                TLS Web Server Authentication, TLS Web Client Authentication
            X509v3 Authority Key Identifier:
                BD:43:1A:9A:9D:06:CC:6A:73:77:25:1B:69:85:B6:7B:45:F1:A2:59
            X509v3 Subject Alternative Name:
                IP Address:127.0.0.1, IP Address:127.0.0.1
    Signature Algorithm: sha256WithRSAEncryption
    Signature Value:
        REDACTED
-----BEGIN CERTIFICATE-----
MIIDQTCCAimgAwIBAgIQXWUJJqmSirjMczp9cK9QBzANBgkqhkiG9w0BAQsFADAr
MRIwEAYDVQQKEwlDb2Nrcm9hY2gxFTATBgNVBAMTDENvY2tyb2FjaCBDQTAeFw0y
MzExMTQyMjE5NTVaFw0yODExMTgyMjE5NTVaMCMxEjAQBgNVBAoTCUNvY2tyb2Fj
aDENMAsGA1UEAxMEbm9kZTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEB
AK6t6tthajEHfnmODgY4oJYLQuq8gNMW439kNTdqGn6mYNbNPtpzVqfeH8F+O24E
+bRo1GHPLZ56ujaLVX5oxOgW8d4Hi60tIXSf1ChsX+RsXAFJWHExEyIC3ljEuh/E
AqXAaD5mVjE4ba0PYvYh7QXr0CD65LHtwoZlua7eiRK8kQCIOAJDZPM8xqPq5h0c
4Te44NY8sYuknPVWOZfejMW+FPlMS4fWqe8CO1Iv9tYFg7greYKFy2wsbs7u6Hfv
+t3qA5FcmjDLyOMQZqgN46MC8uGE0IuKljAyXwgKZhuSBJxtlLSrDm03wksAilaQ
IGgV4/PyQzWisI5CMOP9MtMCAwEAAaNpMGcwDgYDVR0PAQH/BAQDAgWgMB0GA1Ud
JQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAfBgNVHSMEGDAWgBS9QxqanQbManN3
JRtphbZ7RfGiWTAVBgNVHREEDjAMhwR/AAABhwR/AAABMA0GCSqGSIb3DQEBCwUA
A4IBAQB/KfS2c7NrMrwJ1o09ZJe8vbODW8w2oDOiX+KcWx/Ysgublp7fK79q25Y+
jTKXwd4CnfR1XEhkALB2s/kZzUCJJQNbhpfWfsv3D7i8xFx+UVqXrX8/chcWk9pF
hYaRTz1O6SoQb31VRMIPBNp4zIgcoy7/t18YZxGQxZE2RHK/bkouHqgnImgPqr/t
BZMq4jmEySLANRUba8KRHhu2jbMfocgQoUAbob3Xu5bqurKxrSoNeqMxhNKJlZWg
yPQb3mGWDkuU3eLVUGeSmVkvKBGRek/7nCVnwuiAoap7q/FviZFOwpZj3weh7iY8
njPbh704HgY1WA3/+VI9aloX9b8h
-----END CERTIFICATE-----
```

I agree that there may be a feature request here for having the generated TLS certificate use the name specified by `--hostname` or similar.

Reply:
As @sudomateo mentioned, `--insecure` is allowed, but not the default. 

>  I figured that it was intentionally starting the cluster in secure mode even though the multi-node cluster instructions above it used --insecure. 

I wonder what extra benefit this would bring. If the motivation is to make sql console accessible locally (i.e. via `./cockroach sql` directly, rather than via `docker exec -it ./cockroach sql`), the user now is already able to do it by `docker cp` to copy the certs to local file system, and then 
```
./cockroach sql --url="postgresql://root@127.0.0.1:26257/defaultdb?sslcert=certs%2Fclient.root.crt&sslkey=certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=certs%2Fca.crt"
```

Additionally, originally `start-single-node` with docker is intended for tests, so we expect the usage of it doesn't require many customization, and default network setting should be sufficient. 

(FYI I'm currently not on the SQL Foundations team, who owns this part of the codebase, so would like to cc @dikshant and @rafiss in terms of new feature request)

Reply:
@mdlinville bumping this for review this week. The changes are correct in their current state but I would like to resolve this discussion before merging. Let me know what you think.